tutorials-v5/time-evolution/022_v5_paper-smesolve.md

```diff
@@ -0,0 +1,132 @@
+Such a cavity obeys the general stochastic master equation
+
+$d \rho(t) = -i [H, \rho(t)] dt + \mathcal{D}[a] \rho (t) dt + \mathcal{H}[a] \rho dW(t)$
+
+with the Hamiltonian
+
+$H = \Delta a^\dag a$
+
+and the Lindblad dissipator
+
+$\mathcal{D}[a] = a \rho a^\dag - \dfrac{1}{2} a^\dag a \rho - \dfrac{1}{2} \rho a^\dag a$.
+
+The stochastic part
+
+$\mathcal{H}[a]\rho = a \rho + \rho a^\dag - tr[a \rho + \rho a^\dag]$
+
+captures the conditioning of the trajectory through continious monitoring of the operator $a$.
+The term $dW(t)$ is the increment of a Wiener process that obeys $\mathbb{E}[dW] = 0$ and $\mathbb{E}[dW^2] = dt$.
+
+```python
+import numpy as np
+from matplotlib import pyplot as plt
+from qutip import about, coherent, destroy, mesolve, smesolve
+
+%matplotlib inline
+```
+
+## Problem Parameters
+
+```python
+N = 20  # dimensions of Hilbert space
+delta = 10 * np.pi  # cavity detuning
+kappa = 1  # decay rate
+A = 4  # initial coherent state intensity
+```
+
+```python
+a = destroy(N)
+x = a + a.dag()  # operator for expectation value
+H = delta * a.dag() * a  # Hamiltonian
+mon_op = np.sqrt(kappa) * a  # continiously monitored operators
+```
+
+## Solving for the Time Evolution
+
+We calculate the predicted trajectory conditioned on the continious monitoring of operator $a$.
+This is compared to the regular `mesolve()` solver for the same model but without resolving conditioned trajectories.
+
+```python
+rho_0 = coherent(N, np.sqrt(A))  # initial state
+times = np.arange(0, 1, 0.0025)
+num_traj = 500  # number of computed trajectories
+opt = {"dt": 0.00125, "store_measurement": True}
+```
+
+```python
+me_solution = mesolve(H, rho_0, times, c_ops=[mon_op], e_ops=[x])
+```
+
+```python
+stoc_solution = smesolve(
+    H, rho_0, times, sc_ops=[mon_op], e_ops=[x], ntraj=num_traj, options=opt
+)
+```
+
+## Comparison of Results
+
+We plot the averaged homodyne current $J_x = \langle x \rangle + dW / dt$ and the average system behaviour $\langle x \rangle$ for 500 trajectories.
+This is compared with the prediction of the regular `mesolve()` solver that does not include the conditioned trajectories.
+Since the conditioned expectation values do not depend on the trajectories, we expect that this reproduces the result of the standard `me_solve`.
+
+```python
+stoc_meas_mean = np.array(stoc_solution.measurement).mean(axis=0)[0, :].real
+```
+
+```python
+plt.figure()
+plt.plot(times[1:], stoc_meas_mean, lw=2, label=r"$J_x$")
+plt.plot(times, stoc_solution.expect[0], label=r"$\langle x \rangle$")
+plt.plot(
+    times,
+    me_solution.expect[0],
+    "--",
+    color="gray",
+    label=r"$\langle x \rangle$ mesolve",
+)
+
+plt.legend()
+plt.xlabel(r"$t \cdot \kappa$")
+plt.show()
+```
+
+## About
+
+```python
+about()
+```
+
+## Testing
+
+```python
+assert np.allclose(
+    stoc_solution.expect[0] == me_solution.expect[0]
+), "The smesolve and mesolve do not preoduce the same trajectory."
+```
```

Comment:
this test seems, to be failing, i guess this should be `` np.allclose(
    stoc_solution.expect[0] , me_solution.expect[0]
)``

?

Reply:
Yess, thank you. A classic friday afternoon mistake.
But also, smesolve and mesolve do produce slightly different results. Like of the order of 1e-2. Although the general trajecotry is of course the same. So I guess I just put a higher tolerance for np.allclose?

Reply:
Is it maybe instructional to include a plot of this (small) deviation in the notebook? Cause it does also have an unexpected oscillatory behaviour

Reply:
thanks! for the test yeah, maybe just reduce the tolerance. 

does the difference change with ntraj? maybe its some numerical difference in the way to two methods work (e.g.,  I think the smesolve result is pretty sensitive to the choice of dt, while mesolve has more robust dymamic time-step stuff)

Reply:
It does change the shape of the curve a little, but the order of error remains the same. I attach three pictures with ntraj = 50, 500, 5000. So this essentially plots `stoc_solution.expect[0] - me_solution.expect[0]`.
It seems like the curve just gets smoother with higher ntraj count.

### ntraj = 50
![ntraj50](https://github.com/user-attachments/assets/4ac28bc2-bed4-4e3f-93aa-c01a4b81bd19)
### ntraj = 500
![ntraj500](https://github.com/user-attachments/assets/17a02c81-4144-4edf-bcf0-7b5193a825e2)
### ntraj = 5000
![ntraj5000](https://github.com/user-attachments/assets/86a25daa-5177-4267-b6de-ee545a788769)

Reply:
As we thought, the order of the error depends on the timestep size. The oscillatory behaviour remains but for the case below it actually does not seem to grow like in the previous plots.
fyi, in the previous plots i used dt=0.00125, in this plot i used dt=0.00025

![dtsmall](https://github.com/user-attachments/assets/64ead9ac-a22a-458a-a625-1be2fdae7d18)